src/main/java/org/isf/medicalinventory/gui/InventoryEdit.java

```diff
@@ -0,0 +1,2271 @@
+				while (lotListIterator.hasNext()) {
+					Lot lot = lotListIterator.next();
+					inventoryRowTemp = new MedicalInventoryRow(0, lot.getMainStoreQuantity(), lot.getMainStoreQuantity(), null, med, lot);
+					if (!existInInventorySearchList(inventoryRowTemp)) {
+						inventoryRowsList.add(inventoryRowTemp);
+					}
+				}
+			}
+		}
+		return inventoryRowsList;
+	}
+
+	private void addInventoryRow(String code) throws OHServiceException {
+		List<MedicalInventoryRow> inventoryRowsList = new ArrayList<>();
+		List<Medical> medicalList = new ArrayList<>();
+		List<Lot> lots = null;
+		Medical medical = null;
+		MedicalInventoryRow inventoryRowTemp = null;
+		if (code != null) {
+			medical = medicalBrowsingManager.getMedicalByMedicalCode(code);
+			if (medical != null) {
+				medicalList.add(medical);
+			} else {
+				medical = chooseMedical(code);
+				if (medical != null) {
+					medicalList.add(medical);
+				}
+			}
+		} else {
+			medicalList = medicals;
+		}
+		int numberOfMedicalWithoutSameLotAdded = 0;
+		Medical medicalWithLot = null;
+		ListIterator<Medical> medicalListIterator = medicalList.listIterator();
+		while (medicalListIterator.hasNext()) {
+			Medical med = medicalListIterator.next();
+			lots = movStockInsertingManager.getLotByMedical(med, false);
+			if (lots.isEmpty()) {
+				inventoryRowTemp = new MedicalInventoryRow(0, 0.0, 0.0, null, med, null);
+				if (!existInInventorySearchList(inventoryRowTemp)) {
+					inventoryRowsList.add(inventoryRowTemp);
+				} else {
+					int info = MessageDialog.yesNo(null, "angal.inventory.productalreadyexist.msg", med.getDescription());
+					if (info == JOptionPane.YES_OPTION) {
+						inventoryRowsList.add(inventoryRowTemp);
+					}
+				}
+			} else {
+				medicalWithLot = med;
+				ListIterator<Lot> lotListIterator = lots.listIterator();
+				while (lotListIterator.hasNext()) {
+					Lot lot = lotListIterator.next();
+					inventoryRowTemp = new MedicalInventoryRow(0, lot.getMainStoreQuantity(), lot.getMainStoreQuantity(), null, med, lot);
+					if (!existInInventorySearchList(inventoryRowTemp)) {
+						inventoryRowsList.add(inventoryRowTemp);
+						numberOfMedicalWithoutSameLotAdded = numberOfMedicalWithoutSameLotAdded + 1;
+					}
+				}
+			}
+		}
+		if (medicalWithLot != null && numberOfMedicalWithoutSameLotAdded == 0) {
+			int info = MessageDialog.yesNo(null, "angal.inventory.productalreadyexist.msg", medicalWithLot.getDescription());
+			if (info == JOptionPane.YES_OPTION) {
+				inventoryRowTemp = new MedicalInventoryRow(0, 0.0, 0.0, null, medicalWithLot, null);
+				inventoryRowsList.add(inventoryRowTemp);
+			}
+		}
+		for (MedicalInventoryRow inventoryRow : inventoryRowsList) {
+			addMedInRowInInventorySearchList(inventoryRow);
+		}
+		jTableInventoryRow.updateUI();
+	}
+
+	private Medical chooseMedical(String text) throws OHServiceException {
+		Map<String, Medical> medicalMap;
+		medicalMap = new HashMap<String, Medical>();
```

Comment:
```suggestion
		medicalMap = new HashMap<>();
```
I would also suggest merging this line with the line above